packages/legend-application-query/src/components/QueryEditor.tsx

```diff
@@ -189,9 +201,222 @@ const renderQueryEditorHeaderLabel = (
+                              className="table__cell--left"
+                            >
+                              {column}
+                            </th>
+                          ))}
+                        </tr>
+                      </thead>
+                      <tbody>
+                        {(editorStore.isMineOnly
+                          ? editorStore.queries.filter(
+                              (q) => q.isCurrentUserQuery,
+                            )
+                          : editorStore.queries
+                        ).map((q) => (
+                          <tr
+                            key={q.id}
+                            className={clsx(
+                              'query-editor__query-loader__body__table__row',
+                              {
+                                'query-editor__query-loader__body__table__row--selected':
+                                  selectedQueryID === q.id,
+                              },
+                            )}
+                            onClick={(event) => setSelectedQueryID(q.id)}
+                          >
+                            {buildQueryTableRow(q).map((value, columnIdx) => (
+                              // eslint-disable-next-line react/no-array-index-key
+                              <td key={columnIdx} className="table__cell--left">
+                                {value}
+                              </td>
+                            ))}
+                          </tr>
+                        ))}
+                      </tbody>
+                    </table>
+                    <div className="query-editor__query-loader__body__warning">
+                      Show relevant queries by searching names
+                    </div>
+                  </>
+                )}
+                {editorStore.queries.length === 0 && (
+                  <BlankPanelContent>No Query to Preview</BlankPanelContent>
+                )}
+              </>
+            )}
+            {editorStore.isLoadingQueryInProgress && (
+              <>
+                <PanelLoadingIndicator
+                  isLoading={editorStore.isLoadingQueryInProgress}
+                />
+                <BlankPanelContent>loading queries...</BlankPanelContent>
+              </>
+            )}
+          </div>
+          <div className="search-modal__actions">
+            <button className="btn btn--dark" onClick={loadSelectedQuery}>
+              Load
+            </button>
+            <button className="btn btn--dark" onClick={closeQueryImporter}>
+              Close
+            </button>
+          </div>
+        </div>
+      </Dialog>
+    );
+  },
+);
+
 const QueryEditorHeaderContent = observer(() => {
   const editorStore = useQueryEditorStore();
   const applicationStore = useLegendQueryApplicationStore();
+  const loadQuery = (): void => {
+    editorStore.setIsQueryLoaderOpen(!editorStore.isQueryLoaderOpen);
```

Comment:
can we call this `openQueryLoader` also we should be setting `true` or `false` in the input to make it clearer what we are expecting the query loader to do, unless the action is specific to `toggle`